rs/sns/testing/src/main.rs

```diff
@@ -1,22 +1,64 @@
+use std::path::PathBuf;
+
 use clap::Parser;
+use futures::future::join_all;
+use ic_base_types::CanisterId;
+use ic_nervous_system_agent::CallCanisters;
+use ic_nervous_system_integration_tests::pocket_ic_helpers::load_registry_mutations;
+use ic_sns_cli::utils::{dfx_interface, get_agent};
 use ic_sns_testing::nns_dapp::bootstrap_nns;
-use ic_sns_testing::sns::{
-    create_sns, install_test_canister, upgrade_sns_controlled_test_canister, TestCanisterInitArgs,
+use ic_sns_testing::sns::{create_sns, upgrade_sns_controlled_test_canister, TestCanisterInitArgs};
+use ic_sns_testing::utils::{
+    build_ephemeral_agent, get_identity_principal, get_nns_neuron_hotkeys, validate_network,
+    validate_target_canister, NNS_NEURON_ID,
 };
+use icp_ledger::Tokens;
+use k256::elliptic_curve::SecretKey;
 use pocket_ic::PocketIcBuilder;
+use rand::SeedableRng;
+use rand_chacha::ChaChaRng;
 use reqwest::Url;
 
 #[derive(Debug, Parser)]
 struct SnsTestingOpts {
+    #[clap(subcommand)]
+    subcommand: SnsTestingSubCommand,
+}
+
+#[derive(Debug, Parser)]
+enum SnsTestingSubCommand {
+    RunBasicScenario(RunBasicScenarioOpts),
+    NnsInit(NnsInitOpts),
+}
+
+#[derive(Debug, Parser)]
+struct RunBasicScenarioOpts {
+    #[arg(long)]
+    network: String,
+    #[arg(long)]
+    dev_identity: String,
+    #[arg(long)]
+    treasury_identity: String,
+    #[arg(long)]
+    test_canister_id: CanisterId,
+}
+
+#[derive(Debug, Parser)]
+struct NnsInitOpts {
     #[arg(long)]
     server_url: Url,
+    #[arg(long)]
+    state_dir: PathBuf,
+    #[arg(long)]
+    dev_identity: String,
+    #[arg(long)]
+    treasury_identity: String,
 }
 
-#[tokio::main]
-async fn main() {
-    let opts = SnsTestingOpts::parse();
+async fn nns_init(opts: NnsInitOpts) {
     let mut pocket_ic = PocketIcBuilder::new()
         .with_server_url(opts.server_url)
+        .with_state_dir(opts.state_dir.clone())
         .with_nns_subnet()
         .with_sns_subnet()
         .with_ii_subnet()
```

Comment:
Should we have an application subnet for the dapp being decentralized?

Reply:
Hmm, I'm pretty there is `.with_application_subnet()` just below, which is weirdly not present in the diff

Reply:
Sorry just didn't see it due to GitHub hiding that line.